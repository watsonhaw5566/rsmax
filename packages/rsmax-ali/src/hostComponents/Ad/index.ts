import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

export interface AdProps {
  /** 广告展位码 */
  unitId: string;
  /** 否	广告查询成功的回调 */
  onLoad?: (e: any) => void;
  /** 广告查询失败的回调 */
  onError?: (e: any) => void;
}

/**
 * Ad 广告组件
 * https://opendocs.alipay.com/mini/0drov4?pathHash=2b0e72a3
 */
export const Ad: React.ComponentType<AdProps> = createHostComponent<AdProps>('ad');
