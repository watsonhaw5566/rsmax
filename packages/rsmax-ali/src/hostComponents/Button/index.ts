import React from 'react';
import { createHostComponent } from '@rsmax/runtime';

export interface ButtonProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  style?: React.CSSProperties;
  size?: 'default' | 'mini';
  type?: 'primary' | 'default' | 'warn';
  plain?: boolean;
  disabled?: boolean;
  loading?: boolean;
  hoverClassName?: string;
  hoverStartTime?: number;
  hoverStayTime?: number;
  hoverStopPropagation?: boolean;
  formType?: 'submit' | 'reset';
  openType?: 'share' | 'getAuthorize' | 'contactShare' | 'lifestyle' | 'chooseAvatar';
  scope?: 'phoneNumber' | 'userInfo';
  appParameter?: string;
  publicId?: string;
  onTap?: (e: any) => void;
  onClick?: (e: any) => void;
  onGetAuthorize?: (e: any) => void;
  onChooseAvatar?: (e: any) => void;
  onError?: (e: any) => void;
  children?: React.ReactNode;
}

export const Button = createHostComponent<ButtonProps>('button') as React.ComponentType<ButtonProps>;
