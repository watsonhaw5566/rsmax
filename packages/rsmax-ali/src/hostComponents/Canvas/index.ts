import React from 'react';
import { createHostComponent } from '@rsmax/runtime';

export interface CanvasProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  style?: React.CSSProperties;
  className?: string;
  width?: string | number;
  height?: string | number;
  disableScroll?: boolean;
  type?: string;
  onTap?: (e: any) => void;
  onTouchStart?: (e: any) => void;
  onTouchMove?: (e: any) => void;
  onTouchEnd?: (e: any) => void;
  onTouchCancel?: (e: any) => void;
  onLongTap?: (e: any) => void;
  onReady?: (e: any) => void;
}

export const Canvas = createHostComponent<CanvasProps>('canvas') as React.ComponentType<CanvasProps>;
