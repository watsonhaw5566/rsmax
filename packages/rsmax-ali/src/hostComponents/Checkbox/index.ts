import { createHostComponent } from '@rsmax/runtime';

export interface CheckboxProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  name?: string;
  className?: string;
  value?: string;
  checked?: boolean;
  disabled?: boolean;
  color?: string;
  onChange?: (e: any) => void;
}

export const Checkbox = createHostComponent<CheckboxProps>('checkbox') as React.ComponentType<CheckboxProps>;
