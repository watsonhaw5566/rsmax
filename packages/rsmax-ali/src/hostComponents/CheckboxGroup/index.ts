import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

export interface CheckboxGroupProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  name?: string;
  onChange?: (e: any) => void;
}

export const CheckboxGroup = createHostComponent<CheckboxGroupProps>(
  'checkbox-group'
) as React.ComponentType<CheckboxGroupProps>;
