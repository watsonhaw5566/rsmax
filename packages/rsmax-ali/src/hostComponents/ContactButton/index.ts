import React from 'react';
import { createHostComponent } from '@rsmax/runtime';

export interface ContactButtonProps {
  readonly dataset?: DOMStringMap;
  tntInstId: string;
  scene: string;
  size?: string | number;
  color?: string;
  icon?: string;
  alipayCardNo?: string;
}

export const ContactButton = createHostComponent<ContactButtonProps>(
  'contact-button'
) as React.ComponentType<ContactButtonProps>;
