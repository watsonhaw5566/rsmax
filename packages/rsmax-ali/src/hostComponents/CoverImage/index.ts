import React from 'react';
import { createHostComponent } from '@rsmax/runtime';

export interface CoverImageProps {
  readonly dataset?: DOMStringMap;
  className?: string;
  style?: React.CSSProperties;
  src?: string;
  onTap?: (e: any) => void;
}

export const CoverImage = createHostComponent<CoverImageProps>('cover-image') as React.ComponentType<CoverImageProps>;
