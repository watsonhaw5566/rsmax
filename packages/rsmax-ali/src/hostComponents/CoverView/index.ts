import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

export interface CoverViewProps {
  readonly dataset?: DOMStringMap;
  className?: string;
  style?: React.CSSProperties;
  onTap?: (e: any) => void;
}

export const CoverView = createHostComponent<CoverViewProps>('cover-view') as React.ComponentType<CoverViewProps>;
