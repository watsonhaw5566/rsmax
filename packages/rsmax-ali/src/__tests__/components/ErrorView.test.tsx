import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ErrorView } from '../../hostComponents';

describe('ErrorView', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(<ErrorView>text</ErrorView>);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
