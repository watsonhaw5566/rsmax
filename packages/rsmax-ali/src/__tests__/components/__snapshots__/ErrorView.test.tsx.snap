// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorView > render correctly 1`] = `
<error-view
  type="default"
>
  text
</error-view>
`;
